### PR TITLE
Eth -> Method getBlock

### PR DESCRIPTION
**Pull request by kclowes**
### What was wrong?
The Eth module needs to be moved to use Method. Part 3 includes changes to eth_getBlock

Related to Issue #1568

### How was it fixed?

Updated eth.py to use Method for getBlock.

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)
- [x] Squash
- [x] Manually Test

#### Cute Animal Picture

image

### CODE REVIEW

**Comment by kclowes** [comment]
I think the ethpm tests are failing because IPFS is having trouble today 🤷 I'll make sure they're passing before merging.

File: tests/core/gas-strategies/test_time_based_gas_price_strategy.py [hunk 1]
```diff
@@ -35,12 +34,12 @@ def _get_block_by_something(method, params):
                 {'gasPrice': 5},
                 {'gasPrice': 50},
             ],
-            'miner': '0xA',
+            'miner': '0x' + 'AA' * 20,
```
Comment (marcgarreau) [inline tests/core/gas-strategies/test_time_based_gas_price_strategy.py:37]: context for miner hash change?

File: tests/core/middleware/test_filter_middleware.py [hunk 2]
```diff
@@ -38,8 +49,8 @@ def iterator():
 @pytest.fixture(scope='function')
 def result_generator_middleware(iter_block_number):
     return construct_result_generator_middleware({
-        'eth_getLogs': lambda *_: ["middleware"],
-        'eth_getBlockByNumber': lambda *_: type('block', (object,), {'hash': 'middleware'}),
+        'eth_getLogs': lambda *_: FILTER_LOG,
```
Comment (marcgarreau) [inline tests/core/middleware/test_filter_middleware.py:52]: TIL `*_`

File: tests/core/gas-strategies/test_time_based_gas_price_strategy.py [hunk 3]
```diff
@@ -19,8 +19,7 @@ def _get_block_by_something(method, params):
     block_identifier = params[0]
     if (
         block_identifier == 'latest' or
-        block_identifier == '0x0000000000000000000000000000000000000000000000000000000000000005' or
```
Comment (marcgarreau) [inline tests/core/gas-strategies/test_time_based_gas_price_strategy.py:22]: curious about the difference/requirement of prefixed zeros here 🤔

**Review by marcgarreau — APPROVED** [review]
thanks for the walkthrough!

File: tests/core/gas-strategies/test_time_based_gas_price_strategy.py (hunk 1 above)
Comment (kclowes) [reply tests/core/gas-strategies/test_time_based_gas_price_strategy.py:37]: This test provides its own middleware, which did not include any normalization, so previously it didn't matter what got passed back. Now that the normalization is unavoidable, '0xA' throws an error when it tries to convert the address to a checksum address.

File: tests/core/gas-strategies/test_time_based_gas_price_strategy.py (hunk 3 above)
Comment (kclowes) [reply tests/core/gas-strategies/test_time_based_gas_price_strategy.py:22]: This particular block hash just wasn't used in the tests. I could leave it in if you want, I don't have a strong opinion either way. The other block identifier hashes are used with the `getBlockByHash` method below. But I did have to change `5` to `0x5` to account for data normalization that wasn't happening before, but now is.

Comment (marcgarreau) [reply tests/core/gas-strategies/test_time_based_gas_price_strategy.py:22]: 👍 If unused, def leave it out.

File: tests/core/gas-strategies/test_time_based_gas_price_strategy.py (hunk 1 above)
Comment (marcgarreau) [reply tests/core/gas-strategies/test_time_based_gas_price_strategy.py:37]: 👍 thanks for the explanation!

**Pull request merged**